test_scripts/Defects/8_1/3838_3_SDL_does_not_send_UI.ClosePopUp_in_case_two_PerformInteraction_requests.lua

```diff
@@ -0,0 +1,142 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/SmartDeviceLink/sdl_core/issues/3838
+---------------------------------------------------------------------------------------------------
+-- Description: Check that SDL does not send UI.ClosePopUp to HMI in case
+--  mobile App sends two PerformInteraction request with VR_ONLY interaction mode to SDL
+--
+-- Pre-conditions:
+-- 1. SDL, HMI, Mobile session are started
+-- 2. App is registered and activated
+-- 3. ChoiceSets is added
+-- Steps:
+-- 1. Mobile app requests PerformInteraction(VR_ONLY) to SDL
+-- 2. Mobile app requests PerformInteraction(VR_ONLY) one more time while the first one is in progress
+-- 3. HMI responds with 'REJECT' to second UI/VR.PerformInteraction requests
+-- SDL does:
+-- - not send UI.ClosePopUp request to HMI
+-- - sens PerformInteraction response with (resultCode: REJECT, success:false) to mobile App
+-- 4. HMI responds with 'SUCCESS' to first UI/VR.PerformInteraction requests
+-- SDL does:
+-- - sens PerformInteraction response with choiceID from VR response (resultCode: SUCCESS, success:true) to mobile App
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+local utils = require('user_modules/utils')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local function getPromptValue(pText)
+  return {
+    {
+      text = pText,
+      type = "TEXT"
+    }
+  }
+end
+
+local initialPromptValue = getPromptValue(" Make your choice ")
+
+local helpPromptValue = getPromptValue(" Help Prompt ")
+
+local timeoutPromptValue = getPromptValue(" Time out ")
+
+--[[ Local Functions ]]
+local function setChoiceSet(pChoiceIDValue)
+  local temp = {
+    {
+      choiceID = pChoiceIDValue,
+      menuName ="Choice" .. tostring(pChoiceIDValue),
+      vrCommands = {
+        "VrChoice" .. tostring(pChoiceIDValue),
+      }
+    }
+  }
+  return temp
+end
+
+local function createInteractionChoiceSet(pChoiceSetID)
+  local choiceID = pChoiceSetID
+  local cid = common.getMobileSession():SendRPC("CreateInteractionChoiceSet", {
+      interactionChoiceSetID = pChoiceSetID,
+      choiceSet = setChoiceSet(choiceID),
+    })
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", {
+      cmdID = choiceID,
+      type = "Choice",
+      vrCommands = { "VrChoice" .. tostring(choiceID) }
+    })
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+  common.getMobileSession():ExpectResponse(cid, { resultCode = "SUCCESS", success = true })
+end
+
+local function sendPI(pChoiceVR, pExpResult)
```

Comment:
I would suggest sending an expected response  instead of pExpResult
```
local function sendPI(pChoiceVR, pExpResponse)
  local params1 = {
    initialText = "StartPerformInteraction",
    initialPrompt = initialPromptValue,
    interactionMode = "VR_ONLY",
    interactionChoiceSetIDList = { 100 },
    helpPrompt = helpPromptValue,
    timeoutPrompt = timeoutPromptValue,
    timeout = 5000,
    vrHelp = { { text = pChoiceVR, position = 1 } }
  }
  local cid = common.getMobileSession():SendRPC("PerformInteraction", params1)

  common.getMobileSession():ExpectResponse(cid, pExpResponse)
end
```

Reply:
Don't forget edit line (96-97) where running this function and sends arguments 

```
common.run.runAfter(function() sendPI("ChoiceVR1", "SUCCESS") end, 500)
common.run.runAfter(function() sendPI("ChoiceVR2", "REJECTED") end, 1000)

```  
-->
```
common.run.runAfter(function() sendPI("ChoiceVR1", { success = true, resultCode = "SUCCESS" }) end, 500)
common.run.runAfter(function() sendPI("ChoiceVR2", { success = false, resultCode = "REJECTED" }) end, 1000)
```